src/audio/pcm_converter/pcm_converter_generic.c

```diff
@@ -134,24 +134,26 @@ const struct pcm_func_map pcm_func_map[] = {
 #if CONFIG_FORMAT_S16LE
 	{ SOF_IPC_FRAME_S16_LE, SOF_IPC_FRAME_S16_LE, buffer_copy_s16 },
 #endif /* CONFIG_FORMAT_S16LE */
+
 #if CONFIG_FORMAT_S24LE
 	{ SOF_IPC_FRAME_S24_4LE, SOF_IPC_FRAME_S24_4LE, buffer_copy_s32 },
 #endif /* CONFIG_FORMAT_S24LE */
+#if  CONFIG_FORMAT_S24LE && CONFIG_FORMAT_S16LE
+	{ SOF_IPC_FRAME_S16_LE, SOF_IPC_FRAME_S24_4LE, pcm_convert_s16_to_s24 },
+	{ SOF_IPC_FRAME_S24_4LE, SOF_IPC_FRAME_S16_LE, pcm_convert_s24_to_s16 },
+#endif /* CONFIG_FORMAT_S24LE && CONFIG_FORMAT_S16LE */
+
```

Comment:
The "float: " prefix in the patch summary is not correct. This patch has nothing to do with float. The prefixes should refer to the SW module being modified, in this case "pcm_convertor" and style of past commits (to same module) should be used.

Otherwise this patch look ok.